packages/cli/src/cmds/compare-report/ReportSection.ts

```diff
@@ -219,29 +219,50 @@ export default class ReportSection {
       return tokens.join(' ');
     };
 
-    const appmap_url = (revisionName: RevisionName, appmap: AppMap) => {
+    const buildUrlString = (searchParams: Record<string, string>): string => {
+      const url = new URL(options.appmapURL.toString());
+      Object.keys(searchParams).forEach((key) => url.searchParams.append(key, searchParams[key]));
+      return url.toString();
+    };
+
+    const appmap_url = (revisionName: RevisionName, appmap: AppMap): SafeString => {
       let { id } = appmap;
       const path = [revisionName, `${id}.appmap.json`].join('/');
 
-      if (options.appmapURL) {
-        const url = new URL(options.appmapURL.toString());
-        url.searchParams.append('path', path);
-        return new Handlebars.SafeString(url.toString());
-      } else {
-        return new Handlebars.SafeString(path);
-      }
+      let url = path;
+      if (options.appmapURL) url = buildUrlString({ path });
+      return new SafeString(url);
     };
 
     const appmap_diff_url = (appmap: AppMap): SafeString => {
       const path = ['diff', `${appmap.id}.diff.sequence.json`].join('/');
 
+      let url = path;
+      if (options.appmapURL) url = buildUrlString({ path });
+      return new SafeString(url);
+    };
+
+    const appmap_url_with_finding = (
+      revisionName: RevisionName,
+      appmap: AppMap,
+      findingHash: string
+    ) => {
+      let { id } = appmap;
+      const path = [revisionName, `${id}.appmap.json`].join('/');
+
+      let url = path;
       if (options.appmapURL) {
-        const url = new URL(options.appmapURL.toString());
-        url.searchParams.append('path', path);
-        return new Handlebars.SafeString(url.toString());
-      } else {
-        return new Handlebars.SafeString(path);
+        const searchParams = { path } as Record<string, string>;
+        try {
+          const stateObject = { selectedObject: `analysis-finding:${findingHash}` };
+          const state = base64UrlEncode(JSON.stringify(stateObject));
+          searchParams.state = state;
+        } catch (e) {
+          // do not add state
+        }
+        url = buildUrlString(searchParams);
       }
+      return new SafeString(url);
     };
```

Comment:
This looks good. Could you add a test for this to ensure this helper is constructing URLs as expected?

Reply:
Yup, added tests for this helper.